src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -87,6 +105,178 @@
+                )
+            self._authorize_event_sharing = None
+            self._optional_shared_events = []
+            return
+
+        self._manifest_events_definitions = []
+        if not install_method.is_dev_mode:
+            # We can't make any verification for events if we are in prod mode because we do not know event definitions in the manifest file yet.
+            return
+
+        all_events_in_manifest = find_events_in_manifest_file(deploy_root)
+        mandatory_events_in_manifest = find_events_in_manifest_file(
+            deploy_root, mandatory_only=True
+        )
+        self._manifest_events_definitions = self._to_events_definitions(
+            all_events_in_manifest, mandatory_events_in_manifest
+        )
+
+        if self._optional_shared_events:
+            for event in self._optional_shared_events:
+                if event not in all_events_in_manifest:
+                    self._metrics.set_counter(CLICounterField.EVENT_SHARING_ERROR, 1)
+                    raise ClickException(
+                        f"Shared event '{event}' is not found in the manifest file."
+                    )
+
+        if mandatory_events_in_manifest and self._event_sharing_enforced:
+            if self._authorize_event_sharing is None:
+                self._metrics.set_counter(CLICounterField.EVENT_SHARING_WARNING, 1)
+                console.warning(
+                    "WARNING: Mandatory events are present in the manifest file. Automatically authorizing event sharing in dev mode. To suppress this warning, please add 'share_mandatory_events: true' in the application telemetry section."
+                )
+                self._authorize_event_sharing = True
+
+    def _to_events_definitions(
+        self, all_events: List[str], mandatory_events: List[str]
+    ) -> List[Dict[str, str]]:
+        return [
+            {
+                "name": f"SNOWFLAKE${event}",
+                "type": event,
+                "sharing": "MANDATORY" if event in mandatory_events else "OPTIONAL",
+                "status": "DISABLED",
+            }
+            for event in all_events
+        ]
+
+    def should_authorize_event_sharing(
+        self,
+        app_properties: Optional[Dict[str, str]] = None,
+        events_definitions: Optional[List[Dict[str, str]]] = None,
+    ) -> Optional[bool]:
+        if not self._event_sharing_enabled:
+            return None
+
+        if events_definitions is None and self._is_dev_mode:
+            events_definitions = self._manifest_events_definitions
+
+        current_authorization = (
+            app_properties
+            and app_properties.get(AUTHORIZE_TELEMETRY_COL, "false").lower() == "true"
+        )
+
+        mandatory_events_found = False
+        if events_definitions:
+            for event in events_definitions:
+                if event["sharing"] == "MANDATORY":
+                    mandatory_events_found = True
+                    break
```

Comment:
possibly more readable:
`mandatory_events_found = any(event["sharing"] == "MANDATORY" for event in events_definitions)`

Reply:
in new rev, it will be like this.